.gitignore

```diff
@@ -9,7 +9,7 @@
 !.gitattributes
 !.editorconfig
 !.npmrc
-
+!.cspell.config.yaml
```

Comment:
why this?

Reply:
A cspell  config file was generated in my code base so I thought doing that will add it to gitignore

Reply:
Ok, well there is a `cspell.config.yaml` already. This file is used for spell checking in CI. Kind of weird that there also is a `.cspell.config.yaml`.

Anyway, this `.cspell.config.yaml` should not be comitted! I'll figure out why this is happening later.

Reply:
Oh no, there's no `.cspell.config.yaml` it's actually just the `cspell.config.yaml` but I added the dot, thinking it was going to make the `.gitignoee` file notice it

Reply:
But, `cspell.config.yaml` is already not ignored by `.gitignore`!

So I like to setup my `.gitignore` like this...

First i ignore everything that starts with a `.` these are hidden files on linux and this is a nice default behavior.

Then, if there are files or folders that do start with a dot , like the `.vscode` folder, or `.gitignore` file i exclude them from the ignore list. This can be done by prefixing the file pattern with a `!`. So `!.vscode` and `!.gitignore`.

And then I exclude files and folders that do not start with a dot, like `typed` or `node_modules`.

Reply:
My bad, I thought the ! Sign was added to ignore the file. This correction has been noted. Thank you!